pkgs/by-name/ae/aerogramme/package.nix

```diff
@@ -0,0 +1,45 @@
+{
+  lib,
+  rustPlatform,
+  fetchgit,
+  pkg-config,
+  openssl,
+}:
+
+rustPlatform.buildRustPackage {
+  RUSTC_BOOTSTRAP = true;
+
+  pname = "aerogramme";
+  version = "0.3.0";
+
+  src = fetchgit {
+    url = "https://git.deuxfleurs.fr/Deuxfleurs/aerogramme/";
+    hash = "sha256-ER+P/XGqNzTLwDLK5EBZq/Dl29ZZKl2FdxDb+oLEJ8Y=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "imap-codec-2.0.0" = "sha256-o64Q74Q84xLRfU4K4JtcjyS0J8mfoApvUs9siscd0RA=";
+      "imap-flow-0.1.0" = "sha256-IopxybuVt5OW6vFiw/4MxojzaNZrKu2xyfaX6F8IYlA=";
+      "k2v-client-0.0.4" = "sha256-V71FCIsgK3VStFOzVntm8P0vXRobF5rQ74qar+cKyik=";
+      "smtp-message-0.1.0" = "sha256-FoSakm3D1xg1vefLf/zkyvzsij1G0QstK3CRo+LbByE=";
+    };
+  };
```

Comment:
Are you sure that a separate Cargo.lock is required? Did `cargoHash` from [their lockfile](https://git.deuxfleurs.fr/Deuxfleurs/aerogramme/src/branch/main/Cargo.lock) not work?

I ask because there is an [ongoing effort](https://github.com/NixOS/nixpkgs/issues/327063) to reduce the amount of Cargo.lock files in Nixpkgs.

Reply:
I did initially try with cargoHash, but got an error referring to the git dependencies and so to use cargoLock.lockFile instead as they are not currently supported.

If there is another way around this then more than happy to implement it, or can add a comment for clarification.

Reply:
Nope, this is the way to go for now.